openfeature-provider/go/local_resolver_factory.go

```diff
@@ -0,0 +1,249 @@
+package confidence
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"time"
+
+	adminv1 "github.com/spotify/confidence-resolver-rust/openfeature-provider/go/confidence/proto/confidence/flags/admin/v1"
+	resolverv1 "github.com/spotify/confidence-resolver-rust/openfeature-provider/go/confidence/proto/confidence/flags/resolverinternal"
+	iamv1 "github.com/spotify/confidence-resolver-rust/openfeature-provider/go/confidence/proto/confidence/iam/v1"
+	"github.com/tetratelabs/wazero"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+const (
+	defaultPollIntervalSeconds = 300 // 5 minutes - aligned with Java implementation
+	defaultPollLogInterval     = 10  // 10 seconds
+)
+
+// StateProvider is an interface for providing resolver state
+type StateProvider interface {
+	Provide(ctx context.Context) ([]byte, error)
+}
+
+// LocalResolverFactory creates and manages local flag resolvers with scheduled state fetching
+type LocalResolverFactory struct {
+	resolverAPI     *SwapWasmResolverApi
+	stateProvider   StateProvider
+	accountId       string
+	flagLogger      WasmFlagLogger
+	cancelFunc      context.CancelFunc
+	pollInterval    time.Duration
+	logPollInterval time.Duration
+}
+
+// NewLocalResolverFactory creates a new LocalResolverFactory with gRPC clients and WASM bytes
+// If customStateProvider is not nil, it will be used; otherwise creates a FlagsAdminStateFetcher
+// Exposure logging is automatically enabled with gRPC services, disabled with custom StateProvider
+// accountId is required when using customStateProvider, otherwise it's extracted from the token
+func NewLocalResolverFactory(
+	ctx context.Context,
+	runtime wazero.Runtime,
+	wasmBytes []byte,
+	resolverStateServiceAddr string,
+	flagLoggerServiceAddr string,
+	authServiceAddr string,
+	apiClientID string,
+	apiClientSecret string,
+	customStateProvider StateProvider,
+	accountId string,
+) (*LocalResolverFactory, error) {
+	// Get poll interval from environment or use default
+	pollInterval := getPollIntervalSeconds()
+	logPollInterval := time.Duration(defaultPollLogInterval) * time.Second
```

Comment:
`pollInterval` is never used? Looks like the ticker only picks up `logPollInterval` which is not configurable via env var